lib/src/query_parameters/queryable.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:mirrors';
+
+class Queryable {
+  Map<String, dynamic> toQuery() {
+    return _buildMap()
+      ..removeWhere(_removeEmptyOrNulls)
+      ..updateAll(_toURIString);
+  }
+
+  // Load automatically all the declared fields from a query class.
+  Map<String, dynamic> _buildMap() {
+    var instanceMirror = reflect(this);
+    var classMirror = instanceMirror.type;
+
+    return classMirror.declarations.map((key, value) {
+      if (value is VariableMirror) {
+        return MapEntry(MirrorSystem.getName(value.simpleName),
+            instanceMirror.getField(value.simpleName).reflectee);
+      }
+
+      // This will be removed in the toQuery() when `_removeEmptyOrNulls`
+      // is invoked.
+      return MapEntry('key', null);
```

Comment:
So why not return directly `null` instead? Can method return something like:
`Map<String, dynamic> | null`?

Reply:
Not really. I tried some different approaches, but it was not possible, see:

<img width="817" alt="image" src="https://user-images.githubusercontent.com/4116980/209866645-25f214ec-c5fc-40a8-9c73-6cf24f8d33c2.png">

In any case it is not bad, because I already have a handler where I remove the nulls :)